src/Thruway/Role/Subscriber.php

```diff
@@ -128,6 +134,32 @@ protected function processSubscribed(ClientSession $session, SubscribedMessage $
             }
         }
     }
+    
+    /**
+     * Process unsubscribe error
+     * 
+     * @param \Thruway\AbstractSession $session
+     * @param \Thruway\Message\ErrorMessage $msg
+     */
+    protected function processUnsubscribeError(AbstractSession $session, ErrorMessage $msg)
+    {
+        foreach ($this->subscriptions as $key => $subscription) {
+            if ($subscription["unsubscribed_request_id"] === $msg->getErrorRequestId()) {
+                // reject the promise
+                $this->subscriptions[$key]['deferred']->reject($msg);
```

Comment:
This deferred is from the original subscription. It would not be possible to reject this because in order to unsubscribe, you would need the subscribe to resolve first (so that you would have the subscription_id), and at that point, you cannot reject.